blockchain/standalone/subsidy_test.go

```diff
@@ -753,6 +955,76 @@ func TestTotalSubsidyTreasury(t *testing.T) {
 	}
 }
 
+// TestTotalSubsidyDCP0010 ensures the estimated total subsidy produced with the
+// subsidy split defined in DCP0010 matches the expected value.
+func TestTotalSubsidyDCP0010(t *testing.T) {
+	// Locals for convenience.
+	mockMainNetParams := mockMainNetParams()
+	reductionInterval := mockMainNetParams.SubsidyReductionIntervalBlocks()
+	stakeValidationHeight := mockMainNetParams.StakeValidationBeginHeight()
+	votesPerBlock := mockMainNetParams.VotesPerBlock()
+
+	// subsidySum returns the sum of the individual subsidies for the given
+	// height using either the original subsidy split or the modified split
+	// defined in DCP0010.  Note that this value is not exactly the same as the
+	// full subsidy originally used to calculate the individual proportions due
+	// to the use of integer math.
+	cache := NewSubsidyCache(mockMainNetParams)
+	subsidySum := func(height int64, useDCP0010 bool) int64 {
+		work := cache.CalcWorkSubsidyV2(height, votesPerBlock, useDCP0010)
+		vote := cache.CalcStakeVoteSubsidyV2(height, useDCP0010) *
+			int64(votesPerBlock)
+		treasury := cache.CalcTreasurySubsidy(height, votesPerBlock, noTreasury)
+		return work + vote + treasury
+	}
+
+	// Calculate the total possible subsidy.
+	totalSubsidy := mockMainNetParams.BlockOneSubsidy()
+	for reductionNum := int64(0); ; reductionNum++ {
+		// The first interval contains a few special cases:
+		// 1) Block 0 does not produce any subsidy
+		// 2) Block 1 consists of a special initial coin distribution
+		// 3) Votes do not produce subsidy until voting begins
+		if reductionNum == 0 {
+			// Account for the block up to the point voting begins ignoring the
+			// first two special blocks.
+			subsidyCalcHeight := int64(2)
+			nonVotingBlocks := stakeValidationHeight - subsidyCalcHeight
+			totalSubsidy += subsidySum(subsidyCalcHeight, false) * nonVotingBlocks
+
+			// Account for the blocks remaining in the interval once voting
+			// begins.
+			subsidyCalcHeight = stakeValidationHeight
+			votingBlocks := reductionInterval - subsidyCalcHeight
+			totalSubsidy += subsidySum(subsidyCalcHeight, false) * votingBlocks
+			continue
+		}
+
+		// Account for the all other reduction intervals until all subsidy has
+		// been produced.
+		//
+		// Note that this is necessarily an estimate since the exact height at
+		// which DCP0010 should be activated is impossible to know at the time
+		// of this writing.  For testing purposes, the activation height is
+		// estimated to be 638976, or in other words, the 104th reduction
+		// interval
```

Comment:
micro nit: no period at the end of the sentence.
Comment contents reads fine though, unless you want to reiterate that it's _mainnet's_ reduction interval.

Reply:
Updated and clarified that it's for mainnet.